pkg/network/testutil/server.go

```diff
@@ -67,6 +69,13 @@ func StartServerTCP(t testing.TB, ip net.IP, port int) io.Closer {
 	}()
 	<-ch
 
+	require.EventuallyWithT(t, func(collect *assert.CollectT) {
+		conn := PingTCP(t, ip, port)
+		if conn != nil {
+			conn.Close()
+		}
+	}, 10*time.Second, time.Millisecond*10, "timed out waiting for TCP server to come up")
```

Comment:
do these values make sense? if we start a lot of servers this could add up quickly, compared to the 100ms wait that was here before

Reply:
10s is a little long I think. If we fail at this point, the server likely exited or we have some sort of network issue, so we shouldn't have to wait too long. Something like 3s should suffice here. The 10ms polling interval also seems a little aggressive. 500ms?

Reply:
PingTCP doesn't create a new server, it connects to an existing server. 

For the 10 millisecond value, I think that one makes sense. USM folks found with experimentation that most of these `eventually` checks pass on the first or second time, so frequent retries can speed up tests.

The 10 second value maybe makes less sense. I suppose if StartTCPServer was broken consistenty, it would cause integration tests to take a long time to fail. 

I could change it down to 5 or 3, but I do think we want a generous timeout given how we see goroutine weirdnes in the extreme cases in CI.

wdyt?

Reply:
I think I would favor the shortest timeout that consistently improves the reliability of the test since that is most likely to be noticed (ie if we make it too short the test continues to be flakey). a long timeout is the kind of thing that would manifest quietly and add to the already long runtimes of agent CI. can we try 3s?

Reply:
+1